core/payment-driver/gnt/src/gnt/faucet.rs

```diff
@@ -27,17 +49,9 @@ impl EthFaucetConfig {
 
     pub async fn request_eth(&self, address: Address) -> GNTDriverResult<()> {
         log::debug!("request eth");
-        let (resolver, bg) = trust_dns_resolver::AsyncResolver::from_system_conf().unwrap();
-        Arbiter::spawn(bg);
-        let tcp_connector = actix_connect::new_connector(resolver);
-        let client = awc::Client::build()
-            .connector(
-                actix_http::client::Connector::new()
-                    .connector(tcp_connector)
-                    .finish(),
-            )
-            .finish();
+        let client = awc::Client::new();
         let request_url = format!("{}/{}", &self.faucet_address, utils::addr_to_str(address));
+        let request_url = resolve_ip_manually(request_url).await;
```

Comment:
Maybe this could be done during init so that the resolved address is cached?

Reply:
good idea! done :)